circle.yml

```diff
@@ -0,0 +1,38 @@
+general:
+  artifacts:
+    - "tmp"
+
+machine:
+  node:
+    version: 6.1.0
+  ruby:
+    version:
+      2.2.3        
+  environment:
+    CIRCLE_ENV: test
+    RAILS_ENV: test
+  java:
+    version: oraclejdk8
+  services:
+    - redis
+
+dependencies:
+  pre:
+    - wget http://projects.iq.harvard.edu/files/fits/files/fits-1.0.5.zip -O fits-1.0.5.zip
+    - unzip fits-1.0.5.zip -d /tmp/
+    - sudo chmod 755 /tmp/fits-1.0.5/fits.sh
+    - sudo ln -s /tmp/fits-1.0.5/fits.sh /bin/fits.sh
+  override:
+    - bundle check --path=vendor/bundle || bundle install --path=vendor/bundle --jobs=4 --retry=3
+  post:
+    - bundle exec rake hydra:test_server:
+        background: true  
+    - sleep 10 && wget --waitretry=5 --retry-connrefused -v http://localhost:8986/rest
```

Comment:
It looks like this checks for Fedora to be started and assuming Solr succeeds. That's probably true the majority of the time, but in theory starting up Solr could fail as well. Also if for some reason CircleCI is particularly slow, I wonder if we might go past the 5 retries?

What if we created a little script in our horton bin directory that we can call like:
`bin/test_server_wait`

And then populate that script with checks for Fedora and Solr. Something like the little script below. I'm not sure what the wget equivalent of this is, I'm just a bit more familiar with curl.

@jhriv - may actually have a cleaner way of doing this we could try also

```
#!/usr/bin/env sh

printf 'Starting Solr.'
until $(curl --output /dev/null --silent --head --fail http://localhost:8985/solr); do
    printf '.'
    sleep 1
done

printf 'Starting Fedora.'
until $(curl --output /dev/null --silent --head --fail http://localhost:8986/rest); do
    printf '.'
    sleep 1
done
```